Lib/test/test_ctypes/test_python_api.py

```diff
@@ -81,5 +81,36 @@ def test_pyobject_repr(self):
         self.assertEqual(repr(py_object(42)), "py_object(42)")
         self.assertEqual(repr(py_object(object)), "py_object(%r)" % object)
 
+    def test_PyFrame_New_f_back(self):
```

Comment:
This sort of test usually goes in `test_capi`.

First a function to call `PyFrame_New` in `_testcapimodule.c`, then call that function in the test.

So, `pythonapi.PyFrame_New(...)` below would be replaced with `_testcapimodule.new_frame(dummy.__code__, globals(), locals())`.